service-base/src/main/java/org/eclipse/hono/service/amqp/AbstractAmqpEndpoint.java

```diff
@@ -108,13 +109,15 @@ public void onLinkAttach(final ProtonConnection con, final ProtonReceiver receiv
         logger.info("Endpoint [{}] does not support data upload, closing link.", getName());
         receiver.setCondition(ProtonHelper.condition(AmqpError.NOT_IMPLEMENTED, "resource cannot be written to"));
         receiver.close();
+        LinkHelper.freeLinkResources(receiver);
```

Comment:
I would rather have two methods `LinkHelper.closeLink(final ProtonLink link, final boolean releaseResources)` and `LinkHelper.closeLink(final ProtonLink link)` which invokes the former with `true`. In our existing code we would then replace all occurrences of `ProtonLink.close` with the latter variant. Later, when vert.x 3.5.3 is released, we need to update only a single place in our code (LinkHelper), instead of removing the (then obsolete) call to LinkHelper.freeLinkResources`.

Additionally, we also should take care of sessions being closed in a similar (i.e. the same) way. However, for that, we should probably rename `LinkHelper` to `ProtonHelper` or something like that ...